lib/relationLink.ml

```diff
@@ -754,60 +754,127 @@ let print_relation_no_dag conf base po ip1 ip2 =
       print_relation_ok conf base info
   | _ -> Hutil.incorrect_request conf
 
+module RelData = struct
+  type t = {
+    descendants : Gwdb.IperSet.t;
+    depths_1 : Ext_int.Set.t;
+    depths_2 : Ext_int.Set.t;
+  }
+
+  let empty =
+    {
+      descendants = Gwdb.IperSet.empty;
+      depths_1 = Ext_int.Set.empty;
+      depths_2 = Ext_int.Set.empty;
+    }
+
+  let get_depths_1 { depths_1 } = depths_1
+  let get_depths_2 { depths_2 } = depths_2
+
+  let add_depth_1 d ({ depths_1 } as data) =
+    { data with depths_1 = Ext_int.Set.add d depths_1 }
+
+  let add_depth_2 d ({ depths_2 } as data) =
+    { data with depths_2 = Ext_int.Set.add d depths_2 }
+end
+
+let rec max_int_of_list =
+  let rec aux i = function
+    | x :: xs -> if x > i then aux x xs else aux i xs
+    | [] -> i
+  in
+  aux 0
+
+let all_ipers_between ~conf ~base ~store ~start_iper ~end_iper ~max_depths
+    ~get_depths ~add_depth =
+  let max_depth = max_int_of_list max_depths in
+  let max_depths = Ext_int.Set.of_list max_depths in
+  let ancestors = Queue.create () in
+  Queue.push (start_iper, 0) ancestors;
+  let data = Gwdb.Marker.get store start_iper in
+  Gwdb.Marker.set store start_iper (add_depth 0 data);
+  let push_parent a depth pa =
```

Comment:
Could you please use more meaningful identifiers than `a` and `pa`? `a` means *ancestor* and `pa` means *parent*?